File: CS/WindowsFormsApplication1/Form1.cs
using System;
using System.Drawing;
using System.Windows.Forms;
using DevExpress.XtraTab;
using DevExpress.XtraTab.ViewInfo;
using DevExpress.XtraTabbedMdi;

namespace DragDataBetweenMdiTabbedForms
{
	public partial class Form1 : Form
	{
		public Form1()
		{
			InitializeComponent();
		}

		private void Form1_Load(object sender, EventArgs e)
		{
			ChildForm1 childForm1 = new ChildForm1();
			childForm1.MdiParent = this;
			childForm1.Show();

			ChildForm2 childForm2 = new ChildForm2();
			childForm2.MdiParent = this;
			childForm2.Show();
			childForm1.BringToFront();
			
			MdiClient mdiClient = XtraTabbedMdiManager.GetMdiClient(this);
			mdiClient.DragOver += new DragEventHandler(mdiClient_DragOver);
		}

		protected void mdiClient_DragOver(object sender, DragEventArgs e)
		{
			Point p = (xtraTabbedMdiManager1 as IXtraTab).ScreenPointToControl(new Point(e.X, e.Y));
			BaseTabHitInfo hitInfo = xtraTabbedMdiManager1.CalcHitInfo(p);
			if ( hitInfo.HitTest == XtraTabHitTest.PageHeader )
				xtraTabbedMdiManager1.SelectedPage = (XtraMdiTabPage)hitInfo.Page;
		}
	}
}

File: VB/WindowsFormsApplication1/Form1.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Drawing
Imports System.Windows.Forms
Imports DevExpress.XtraTab
Imports DevExpress.XtraTab.ViewInfo
Imports DevExpress.XtraTabbedMdi

Namespace DragDataBetweenMdiTabbedForms
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub Form1_Load(ByVal sender As Object, ByVal e As EventArgs) Handles MyBase.Load
			Dim childForm1 As New ChildForm1()
			childForm1.MdiParent = Me
			childForm1.Show()

			Dim childForm2 As New ChildForm2()
			childForm2.MdiParent = Me
			childForm2.Show()
			childForm1.BringToFront()

			Dim mdiClient As MdiClient = XtraTabbedMdiManager.GetMdiClient(Me)
			AddHandler mdiClient.DragOver, AddressOf mdiClient_DragOver
		End Sub

		Protected Sub mdiClient_DragOver(ByVal sender As Object, ByVal e As DragEventArgs)
			Dim p = CType(Me.XtraTabbedMdiManager1, DevExpress.XtraTab.IXtraTab).ScreenPointToControl(New Point(e.X, e.Y))
			Dim hitInfo As BaseTabHitInfo = xtraTabbedMdiManager1.CalcHitInfo(p)
			If hitInfo.HitTest = XtraTabHitTest.PageHeader Then
				xtraTabbedMdiManager1.SelectedPage = CType(hitInfo.Page, XtraMdiTabPage)
			End If
		End Sub
	End Class
End Namespace
